internal/controller/nonadminbackup_controller.go

```diff
@@ -640,6 +643,42 @@ func (r *NonAdminBackupReconciler) createVeleroBackupAndSyncWithNonAdminBackup(c
 			Spec: *backupSpec,
 		}
 
+		if backupSpec.StorageLocation != constant.EmptyString {
```

Comment:
move this up to line 635 to be alongside `backupSpec.IncludedNamespaces = []string{nab.Namespace}`

Reply:
We need line 636 before this if, because on line 679 we use object from 636:
```
			veleroBackup.Spec.StorageLocation = veleroBackupStorageLocation.Name
```

Reply:
Actually I can assign to pointer directly, so will move that to the 635

Reply:
Should be fixed now.